Hrt 45 preview page

lucycy98 opened the pull request:
### Purpose

Created a preview page (/Preview) which displays sliders and user explanation for each dimension.

[HRT-45](https://softeng761team5.atlassian.net/browse/HRT-45)

### Approach

- Refactored DisplayCards.tsx so that the header can be reused in Preview.tsx
- Created a dimension component (components/Dimension.tsx) representing a dimension card (with props: dimension value, scale, user explanation)
- Created preview page (pages/Preview.tsx) with dummy data for dimensions
- Mapped each dummy dimension to a dimension component, passing in the values
- using Antd, created sliders and set default values
- Added back and save navigation buttons

### Does this PR introduce a breaking change?

- [] Yes
- [x] No

### Screenshots before and after this change (required for UI changes)

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

No preview page.

#### Screenshot(s) of UI after this PR

![image](https://user-images.githubusercontent.com/34847224/92765516-83a85c80-f3e9-11ea-98e6-1cbaabccff9f.png)
![image](https://user-images.githubusercontent.com/34847224/92765595-99b61d00-f3e9-11ea-980a-29044bbe08c8.png)

lucycy98 committed:
refactor(header.tsx): created new header component to be shared

lucycy98 committed:
feat(preview.tsx & dimension.tsx & css): added preview page with dimension cards using props

lucycy98 committed:
feat(dimension.tsx & css): added sliders for dimensions with ant styling

lucycy98 committed:
feat(preview.tsx): added back and save navigation buttons to preview page

gargi-bansal commented on src/styles/custom-antd.less:
```diff
@@ -57,22 +57,22 @@
 // @progress-circle-text-font-size: 1em;
 
 // Slider
```
I forgot to mention that you don't need to fill all the commented variables in, only the ones you need :)

gargi-bansal commented on src/styles/custom-antd.css:
```diff
@@ -15166,7 +15166,7 @@ textarea.ant-input-number {
   color: #212121;
   line-height: 64px;
-  background: linear-gradient(to bottom right, #2e3458 0%, #212121 100%);
+  background: #001529;
```
It was mentioned in the .less file but all instances of #001529 in the css file were changed to "linear-gradient(to bottom right, #2e3458 0%, #2121 100%)" to get that dark gradient background

gargi-bansal commented on src/styles/custom-antd.less:
```diff
@@ -8,9 +8,9 @@
-
 @body-background: #f6f7fc;
 @component-background: #ffffff;
+@dark-background: #001529;
```
Do we use this colour anywhere in the designs?

gargi-bansal commented on src/components/Dimension.tsx:
```diff
@@ -0,0 +1,26 @@
+import React, { useState } from "react";
```
unused import statement here :)

gargi-bansal commented on src/components/Header.tsx:
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from "react";
```
another unused import statement :)

gargi-bansal reviewed (changes requested):
Hey Lucy, you've done a really good job with the Preview stuff. Just a few minor things:
- Is there a reason the slider dot is vertically slightly off-centre and can it be fixed? (My OCD getting to me haha)
- You've forgotten to add the explanation/reason text boxes in each of the dimensions so that will need to be added in. They're the text under the slider in the designs.
- Also, it may be best to test the page with an overflow of dimensions (i.e. more than 4 dimensions) and make sure that the dimensions scroll but that the Back and Save buttons stay floating at the bottom as per the hi-fi designs. This is probably one of the key uses of the preview page. There's also a white-transparent gradient at the bottom of the page behind the buttons on the designs that is missing.
- Oh furthermore, I'm not sure why CSS is doing this but we need to have unique class names. At the moment the Header.css and Homepage.css both have class names "Logo-Title". Currently, the Header one is overriding the Homepage one (HeaRT text on Homepage should be broken if you have a look at it). I'll make a ticket for this to be fixed in another issue, but for now can you change the header "Logo-Title" class name to something else.
Thanks!! :)

gargi-bansal reviewed (commented):
I just realised, but isn't this feature also supposed to add a slider to the Display Cards page? I'm pretty sure Preview isn't in this sprint backlog and that this ticket was only related to the Display Cards page.

lucycy98 commented:
> I just realised, but isn't this feature also supposed to add a slider to the Display Cards page? I'm pretty sure Preview isn't in this sprint backlog and that this ticket was only related to the Display Cards page.

@gargi-bansal from my understanding (from the UI mock ups and Claire's communications), in the Display Cards page, the user only needs to select one statement out of two, and this translates to a particular dimension (i.e if they select statement 1, then the dimension leans more towards fixed and so that is where the slider position is initially placed). In the preview page, the user can adjust that slider position, if they feel like its more of a continuum. 

The Preview and save ticket in the other sprint [HRT-19](https://softeng761team5.atlassian.net/browse/HRT-19) is more so about saving the slider locations, maybe generating an image or just saving to db. therefore I think this ticket [HRT-34](https://softeng761team5.atlassian.net/browse/HRT-34) is redundant.

lucycy98 commented:
> Hey Lucy, you've done a really good job with the Preview stuff. Just a few minor things:
> 
> * Is there a reason the slider dot is vertically slightly off-centre and can it be fixed? (My OCD getting to me haha)
> * You've forgotten to add the explanation/reason text boxes in each of the dimensions so that will need to be added in. They're the text under the slider in the designs.
> * Also, it may be best to test the page with an overflow of dimensions (i.e. more than 4 dimensions) and make sure that the dimensions scroll but that the Back and Save buttons stay floating at the bottom as per the hi-fi designs. This is probably one of the key uses of the preview page. There's also a white-transparent gradient at the bottom of the page behind the buttons on the designs that is missing.
> * Oh furthermore, I'm not sure why CSS is doing this but we need to have unique class names. At the moment the Header.css and Homepage.css both have class names "Logo-Title". Currently, the Header one is overriding the Homepage one (HeaRT text on Homepage should be broken if you have a look at it). I'll make a ticket for this to be fixed in another issue, but for now can you change the header "Logo-Title" class name to something else.
>   Thanks!! :)

Great advice I'll get onto that ASAP! One thing is the user explanation textbox is another user story [HRT-16](https://softeng761team5.atlassian.net/browse/HRT-16) so I have left it.

lucycy98 committed:
fix(custom-antd.less & css): centred handle position, changed slider colours to default

lucycy98 committed:
feat(dimension.tsx & css): added user explanations to preview cards

lucycy98 commented:
new fixes
- created new ticket for preview chart
- removed unused import statements
- changed colours to match the blue from the UI diagrams
- added user explanations to cards
- centered slider handle position
- added more dimension cards to test flow

![image](https://user-images.githubusercontent.com/34847224/92834946-bde81d00-f42e-11ea-8d1a-8944c27f1f07.png)
![image](https://user-images.githubusercontent.com/34847224/92834986-cb9da280-f42e-11ea-869b-4b92ebd7f574.png)

gargi-bansal reviewed (changes requested):
Hey, new changes look good! Just want to clarify a few more things and ask questions:
- The explanation text in the preview page isn't editable and it should be (this is the original plan but open to discussion on this)
- Is the slider on the Display Cards page going to be added in a different PR?
- The header background color is still a solid #001529 and should be a gradient as per the original designs (I think I mentioned the gradient declaration in the last comment). You will have to replace all instances of #001529 in the custom-antd.css with the gradient declaration.
- Could the vertical spacing between the slider and the dimension labels (e.g. active/fixed) be reduced, please. The grouping looks a bit off because they're very far apart
- Back/Save buttons should be floating on the bottom left and right of the page. There should be a white-transparent gradient behind it (as in the hi-fis). At the moment you have to scroll all the way to the bottom. 
If you need any help let me know :) It is a fair amount of changes

lucycy98 committed:
fix(navigation.css): made buttons fixed

lucycy98 committed:
fix(custom-ant.css): replaced colours manually

lucycy98 commented:
> Hey, new changes look good! Just want to clarify a few more things and ask questions:
> 
> * The explanation text in the preview page isn't editable and it should be (this is the original plan but open to discussion on this)
the UI mock ups dont have an editable field for this, but we can add it to another user story as this one is simply just displaying it.
> * Is the slider on the Display Cards page going to be added in a different PR?
yeah, this is just the preview page.
> * The header background color is still a solid #001529 and should be a gradient as per the original designs (I think I mentioned the gradient declaration in the last comment). You will have to replace all instances of #001529 in the custom-antd.css with the gradient declaration.
done
> * Could the vertical spacing between the slider and the dimension labels (e.g. active/fixed) be reduced, please. The grouping looks a bit off because they're very far apart
done
> * Back/Save buttons should be floating on the bottom left and right of the page. There should be a white-transparent gradient behind it (as in the hi-fis). At the moment you have to scroll all the way to the bottom.
haven't done this before and I'm running low on time, so would be good if you someone could help with this, I think we could add it to another user story though. probably dont want too much delay as it is a blocked for the display card user story.
>   If you need any help let me know :) It is a fair amount of changes
![image](https://user-images.githubusercontent.com/34847224/92845713-656b4c80-f43b-11ea-9042-5d5a81c89053.png)

TCHE614 committed:
refactor(navigation.css): added in a gradient for back and save button

TCHE614 commented:
Added in the gradient for the footer, it is currently in normal CSS since that div isn't in antd. I was wondering if you guys think its better to create an antd for it or is this implementation fine for now?

gargi-bansal committed:
feat(front-end): make margins on pages consistent

Make footer/header margins consistent from page to page

gargi-bansal reviewed (dismissed):
Made a small styling fix otherwise looks good to me
<img width="750" alt="Screen Shot 2020-09-11 at 8 53 19 PM" src="https://user-images.githubusercontent.com/42624632/92896917-e2fd7f80-f470-11ea-9281-63e9448114ec.png">
<img width="750" alt="Screen Shot 2020-09-11 at 8 53 29 PM" src="https://user-images.githubusercontent.com/42624632/92896940-e7c23380-f470-11ea-8ef3-ea6d0157f239.png">

gargi-bansal commented:
> Added in the gradient for the footer, it is currently in normal CSS since that div isn't in antd. I was wondering if you guys think its better to create an antd for it or is this implementation fine for now?

Should be fine as is unless/until we actually want to use a footer component :)

preetpatel reviewed (approved):
Had a few small UI stuff but meh - happy weekend :)
